packages/editor/src/client/panel/mapLeafletEditPanel.tsx

```diff
@@ -0,0 +1,213 @@
+        title="Edit map markers"
+        leftChildren={
+          <NavigationButton
+            icon={MaterialIconClose}
+            label="Close"
+            onClick={() => onClose?.(items)}
+          />
+        }
+      />
+      <PanelSection>
+        <ListInput
+          value={items}
+          onChange={items => setItems(items)}
+          defaultValue={defaultItemValue}>
+          {props => <MapLeafletItems {...props} />}
+        </ListInput>
+      </PanelSection>
+    </Panel>
+  )
+}
+
+export function MapLeafletItems({value, onChange}: FieldProps<MapLeafletItem>) {
+  const {address, lat, lng, title, description} = value
+
+  return (
+    <>
+      <Box display="flex" flexDirection="column">
+        <AddressInput
+          label="Address"
+          value={[{address: address, lat: lat, lng: lng}]}
+          marginBottom={Spacing.ExtraSmall}
+          onChange={markerPoints => {
+            if (markerPoints !== undefined && markerPoints.length > 0) {
+              onChange({
+                ...value,
+                address: markerPoints[1].address,
+                lat: markerPoints[1].lat,
+                lng: markerPoints[1].lng
+              })
+            }
+          }}
+        />
+        <TextInput
+          marginBottom={Spacing.ExtraSmall}
+          type="number"
+          label="Latitude"
+          value={lat}
+          onChange={e => onChange({...value, lat: parseInt(e.target.value)})}
+          required
+        />
+        <TextInput
+          marginBottom={Spacing.ExtraSmall}
+          type="number"
+          label="Longitude"
+          value={lng}
+          onChange={e => onChange({...value, lng: parseInt(e.target.value)})}
+          required
+        />
+        <TextInput
+          marginBottom={Spacing.ExtraSmall}
+          label="Title"
+          value={title}
+          onChange={e => onChange({...value, title: e.target.value})}
+        />
+        <TextInput
+          marginBottom={Spacing.ExtraSmall}
+          label="Description"
+          value={description}
+          onChange={e => onChange({...value, description: e.target.value})}
+        />
+      </Box>
+    </>
+  )
+}
+
+export interface AddressInputProps extends MarginProps {
+  label?: string
+  description?: string
+  value: MarkerPoint[]
+  onChange(address?: MarkerPoint[]): void
+}
+
+export function AddressInput(props: AddressInputProps) {
+  return (
+    <AutocompleteInput
+      {...props}
+      label={props.label}
+      valueToChipData={markerPoint => ({
+        id: nanoid(),
+        label: markerPoint.address
+      })}>
+      {propsAutocompleteInput => <AddressInputList {...propsAutocompleteInput} />}
+    </AutocompleteInput>
+  )
+}
+
+function AddressInputList({
+  isOpen,
+  inputValue,
+  highlightedIndex,
+  getItemProps,
+  getMenuProps
+}: AutocompleteInputListProps) {
+  const [items, setItems] = useState<MarkerPoint[]>([])
+  const api = 'ab48a007c6dd4906adcd2871a5deaebe'
```

Comment:
never commit api keys to the repository. The belong to the `.env`

Reply:
Done